nimiqclient/nimiq_client.py

```diff
@@ -0,0 +1,560 @@
+        return self.call("minFeePerByte", params)
+
+    def mining(self, state = None):
+        """
+        Returns true if client is actively mining new blocks.
+        When no parameter is given, it returns the current state.
+        When a value is given as parameter, it sets the current state to that value.
+        :param state: The state to be set.
+        :return: True if the client is mining, otherwise False.
+        """
+        params = []
+        if state != None:
+            params.append(state)
+        return self.call("mining", params)
+
+    def peerCount(self):
+        """
+        Returns number of peers currently connected to the client.
+        :return: Number of connected peers.
+        """
+        return self.call("peerCount", [])
+
+    def peerList(self):
+        """
+        Returns list of peers known to the client.
+        :return: The list of peers.
+        """
+        return self.call("peerList", [])
+
+    def peerState(self, address, command = None):
+        """
+        Returns the state of the peer.
+        When no command is given, it returns peer state.
+        When a value is given for command, it sets the peer state to that value.
+        :param address: The address of the peer.
+        :param command: The command to send.
+        :return: The current state of the peer.
+        """
+        params = []
+        params.append(address)
+        if command != None:
+            params.append(command)
+        return self.call("peerState", params)
+
+    def pool(self, address = None):
+        """
+        Returns or sets the mining pool.
+        When no parameter is given, it returns the current mining pool.
+        When a value is given as parameter, it sets the mining pool to that value.
+        :param address: The mining pool connection string ("url:port") or boolean to enable/disable pool mining.
+        :return: The mining pool connection string, or None if not enabled.
+        """
+        params = []
+        if address != None:
+            params.append(address)
+        return self.call("pool", params)
+
+    def poolConfirmedBalance(self):
+        """
+        Returns the confirmed mining pool balance.
+        :return: The confirmed mining pool balance (in smallest unit).
+        """
+        return self.call("poolConfirmedBalance", [])
+
+    def poolConnectionState(self):
+        """
+        Returns the connection state to mining pool.
+        :return: The mining pool connection state.
+        """
+        return self.call("poolConnectionState", [])
+
+    def sendRawTransaction(self, transaction):
+        """
+        Sends a signed message call transaction or a contract creation, if the data field contains code.
+        :param transaction: The hex encoded signed transaction
+        :return: The Hex-encoded transaction hash.
+        """
+        return self.call("sendRawTransaction", [transaction])
+
+    def sendTransaction(self, transaction):
```

Comment:
The send method expects a fully encoded TX data object. I think it's going to be hard to use this RPC client if there is no support in creating this objects. See https://github.com/nimiq-community/java-client/blob/master/src/main/java/com/nimiq/NimiqClient.java for an example.

Reply:
Similar things might apply for other methods.

Reply:
Ok, digging through the code (see comments below also) it seems to be a JSON object being passed here. So the doc is wrong. But let's please confirm. I havn't used Python much, but is it possible to define some from of struct that fixates the attributes needed for a TX and other important objects? How would somebody using this lib know how to format a TX to use the `sentTransaction` method?

Reply:
It seems the doc is wrong, I'll fix it and double check all other docs, it seems an issue related with copy pasting the comments between methods. I'll also add the structures defined in the RPC API like for instance [OutgoingTransaction](https://github.com/nimiq/core-js/wiki/JSON-RPC-API#outgoingtransaction), so that have all attributes fixated in data structures.

Reply:
The doc is [fixed now](https://github.com/rraallvv/python-client/blob/master/nimiqclient/nimiq_client.py#L1190). All methods that typically receive or return json data structures were refactored to use data structures with predefined fixates attributes instead.